magma/array.py

```diff
@@ -325,7 +338,7 @@ def as_list(self):
     def wire(i, o, debug_info):
         # print('Array.wire(', o, ', ', i, ')')
 
-        if not isinstance(o, ArrayType):
+        if not isinstance(magma_value(o), ArrayType):
```

Comment:
should we just do `o = magma_value(o)` at the top? For example, below we do `o.N`...what if `o` is protocol type there?

Reply:
called magma_value before calling wire.